pycode/memilio-epidata/memilio/epidata_test/test_epidata_get_population_data.py

```diff
@@ -142,198 +103,79 @@ class Test_getPopulationData(fake_filesystem_unittest.TestCase):
-
-        self.assertEqual(len(os.listdir(directory)), 2)
-        self.assertEqual(os.listdir(directory), [
-                         d.filename + ".json", d.filename_out + ".json"])
-
-        f_path = os.path.join(directory, d.filename_out + ".json")
-        f = open(f_path, "r")
-        self.assertEqual(f.read(), self.test_string2r)
-
-    def test_gpd_read_data_full(self):
-
-        [read_data, file_format, out_folder, no_raw] = [
-            True, "json", self.path, False]
-
-        directory = os.path.join(out_folder, 'Germany/')
-        gd.check_dir(directory)
-
-        file1 = "FullDataB.json"
-        file2 = "FullDataL.json"
-
-        file_out1 = "PopulStates.json"
-        file_out2 = "PopulCounties.json"
-
-        with open(os.path.join(directory, file1), 'w') as f:
-            f.write(self.test_string1)
-        with open(os.path.join(directory, file2), 'w') as f:
-            f.write(self.test_string2)
-
-        gpd.get_population_data(read_data, file_format, out_folder)
-
-        self.assertEqual(len(os.listdir(directory)), 3)
-        self.assertEqual(os.listdir(directory), [file1, file2, file_out1])
-
-        f_path = os.path.join(directory, file_out1)
-        f = open(f_path, "r")
-        self.assertEqual(f.read(), self.test_string1r)
+    data = np.zeros((5, len(columns_gender)))
+    for i in range(len(data)):
+        data[i, 0] = 1001 + i
+        data[i, 1] = 44 * (i+1)
+        data[i, 2:] = 2 * (i+1)
+    test_current_population_gender_result = pd.DataFrame(
+        data, columns=columns_gender)
+    test_current_population_gender_result = test_current_population_gender_result.astype(
+        'int64')
 
+    def setUp(self):
+        self.setUpPyfakefs()
     def test_get_new_counties(self):
         test = gpd.get_new_counties(self.test_old_counties)
         self.assertTrue(np.array_equal(test, self.test_new_counties))
+    
+    @patch('memilio.epidata.getPopulationData.load_population_data',
+           return_value=(test_counties, test_zensus, test_reg_key))
+    def test_get_population(self, mock_data):
 
-    @patch('memilio.epidata.getPopulationData.load_age_population_data', return_value=(test_counties, test_reg_key, test_zensus))
-    def test_get_age_population(self, mock_data):
+        [read_data, file_format, out_folder, no_raw, split_gender,
+         merge_eisenach] = [True, "json", self.path, False, False, False]
 
-        gpd.get_age_population_data(False, 'json', self.path)
+        gpd.get_population_data(
+            read_data, file_format, out_folder, no_raw, split_gender,
+            merge_eisenach)
 
         test_df = pd.read_json(os.path.join(
-            self.path, 'Germany/', 'county_population.json'))
-        pd.testing.assert_frame_equal(test_df, self.test_population_result)
+            self.path, 'Germany/', 'county_current_population_dim401.json'))
+        test_df = test_df.drop(
+            test_df[test_df[dd.EngEng['population']] == 0].index)
```

Comment:
Why are there zero lines that are dropped here?

Reply:
This is a test for only 4 counties. All other counties can be removed for the test.